### PR TITLE
Add Africa's talking adapter

### PR DESCRIPTION
**Pull request by andela-mmakinde**
**What does this PR do?**

Add Africa's talking sms adapter to the application.

**Description of Task to be completed?**
Add functionality for SMS to work with Africa's Talking. issue https://github.com/CodeForAfricaLabs/HealthTools.API/issues/56

**How should this be manually tested?**
- Create an account with Africa's talking, create a team and an app with them and get an API key.
- Get a shortcode users can send messages to on Africa's talking.
- Add an sms callback url `<host>/sms/africastalking` on Africa's talking.
- Export the variables SMS_AFRICASTALKING_USER and SMS_AFRICASTALKING_KEY.
- Start the app.
- Send a request to the shortcode registered and a response will be sent to your mobile number.

**Screenshots (if appropriate)**
screen_shot_2017-11-15_at_11 51 26_am

bb99352e-ea63-429f-b965-68901c91d5f4

### CODE REVIEW

File: healthtools/search/nurses.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def get_nurses_from_nc_registry(query):
     nurses = {'hits': [], 'total': 0}
     try:
         response = requests.get(url)
-        if 'No results' in response.content:
+        if b'No results' in response.content:
```
Comment (DavidLemayian) [inline healthtools/search/nurses.py:24]: What's the "b" for?

File: healthtools/settings.py [hunk 2]
```diff
@@ -44,6 +44,10 @@
 SMS_MTECH_PASS = os.getenv('HTOOLS_SMS_MTECH_PASS')
 SMS_MTECH_SHORTCODE = os.getenv('HTOOLS_SMS_MTECH_SHORTCODE', '22495')
 
+AFRICASTALKING = {
+    "SMS_AFRICASTALKING_USER": os.getenv("SMS_AFRICASTALKING_USER"),
+    "SMS_AFRICASTALKING_KEY": os.getenv("SMS_AFRICASTALKING_KEY"),
+}
```
Comment (DavidLemayian) [inline healthtools/settings.py]: Similar to MTECH, some 3 things here:

1. Let's have these as their own variables
2. env variables must start with `HTOOLS_`
3. Use single quotes

File: healthtools/sms/__init__.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def process_sms(args, adapter='mtech'):
     # TODO: Track event SMS RECEIVED here
 
     result, doc_type = run_query(msg)
-
+
```
Comment (DavidLemayian) [inline healthtools/sms/__init__.py]: PEP 8 check.

File: healthtools/views/sms.py [hunk 4]
```diff
@@ -10,7 +10,14 @@
...
 @blueprint.route('/sms/<adapter>', methods=['GET', 'POST'])
 def index(adapter='mtech'):
-    result = process_sms(request.args, adapter)
+    arguments = request.args
+    if adapter == "africastalking":
+        arguments = {
+            'phoneNumber': request.values.get('from'),
+            'message': request.values.get('text')
+        }
+
```
Comment (DavidLemayian) [inline healthtools/views/sms.py]: This would be done in the adapter itself similar to how we've achieved Twilio.

File: runtime.txt [hunk 5]
```diff
@@ -1 +1 @@
-python-2.7.13
\ No newline at end of file
+python-3.6.2
```
Comment (DavidLemayian) [inline runtime.txt:1]: Is this correct?

**Comment by DavidLemayian** [comment]
What's the `src/healthtools` submodule for?

File: healthtools/search/nurses.py (hunk 1 above)
Comment (andela-mmakinde) [reply healthtools/search/nurses.py:24]: It turns the text in front of it into a bytes literal.
After updating the app to python3, it throws an error `if 'No results' in response.content:
TypeError: a bytes-like object is required, not 'str'`

File: runtime.txt (hunk 5 above)
Comment (andela-mmakinde) [reply runtime.txt:1]: @andela-ookoro will add this changes in his PR that upgrades to python3

**Pull request closed without merging**